pkg/collector/corechecks/servicediscovery/module/envs.go

```diff
@@ -153,3 +186,168 @@ func getEnvs(proc *process.Process) (map[string]string, error) {
+		env:     make([]byte, 0, defSizeEnvBuf),
+	}
+}
+
+// Finish closes an open file
+func (es *EnvScanner) Finish() {
+	if es.file != nil {
+		es.file.Close()
+	}
+}
+
+// hashBytes return hash value of a bytes array using FNV-1a hash function
+func hashBytes(b []byte) uint64 {
+	h := fnv.New64a()
+	h.Write(b)
+	return h.Sum64()
+}
+
+// addEnvToMapIfTarget add env. variable to map if it matches target or target map is empty.
+func (es *EnvScanner) addEnvToMapIfTarget() {
+	if len(es.envs) == maxSizeEnvsMap {
+		es.err = fmt.Errorf("proc environment scanner can't add more than max (%d)", maxSizeEnvsMap)
+		return
+	}
+	eq := bytes.IndexByte(es.env, '=')
+	if eq == -1 {
+		return
+	}
+	if len(es.targets) > 0 {
+		h := hashBytes(es.env[:eq])
+		_, exists := es.targets[h]
+		if exists {
+			name := string(es.env[:eq])
+			es.envs[name] = string(es.env[eq+1:])
+		}
+	} else {
+		name := string(es.env[:eq])
+		es.envs[name] = string(es.env[eq+1:])
+	}
+}
+
+// ScanFile reads a text file in chunks and extracts null-terminated strings,
+// extracts the name and value of environment variables and stores them in a map.
+// +-------------------------------+---------------+------------+-------------------+
+// | read fragment may include     | end of string | equal sign | action to do      |
+// |-------------------------------|---------------|------------|-------------------|
+// | ^^^^^^[NULL]^^^^=^^^^^[NULL]  |     v         |     v      | save env.var      |
+// | ^^^^^^[NULL]^^^^^^^^^^^^^^^^  |     v         |     -      | save env.var      |
+// | ^^^^^^^^^^^^^^^^=^^^^^^^^^^^  |     -         |     v      | append to env.var |
+// | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^  |     -         |     -      | append to env.var |
+// +-------------------------------+---------------+------------+-------------------+
+func (es *EnvScanner) ScanFile() error {
+	if len(es.envs) == maxSizeEnvsMap {
+		es.err = fmt.Errorf("proc environment scanner reached maximum entries (%d)", maxSizeEnvsMap)
+		return es.err
+	}
+	es.end, es.err = es.file.Read(es.buffer)
+	if es.end <= 0 {
+		return io.EOF
+	}
+	if es.err != nil {
+		return es.err
+	}
+	cursor := 0 // indicates the beginning of the next string
+
+	for cursor < es.end {
+		strLen := bytes.IndexByte(es.buffer[cursor:], 0)
+		strLen = min(strLen, es.end-cursor) // consider only bytes up to the end of the data end
+
+		if strLen >= 0 {
+			// zero length means NULL was encountered immediately.
+			if strLen > 0 {
+				es.env = append(es.env, es.buffer[cursor:cursor+strLen]...)
+			}
+			es.addEnvToMapIfTarget()
+			if es.err != nil {
+				return es.err
+			}
+			// reset processed env. var.
+			es.env = es.env[:0]
+			cursor += strLen + 1
+		} else {
+			es.env = append(es.env, es.buffer[cursor:es.end]...)
+			break
+		}
+	}
+
+	return nil
+}
+
+// getEnvironPath return path '/proc/<PID>/environ'
+func getEnvironPath(proc *process.Process) string {
+	hostProc := os.Getenv("HOST_PROC")
```

Comment:
`kernel.HostProc()` can be used instead.

Reply:
I've changed code to use kernel.HostProc